sectionproperties/pre/sections.py

```diff
@@ -1,259 +1,587 @@
-    :cvar facets: List of point index pairs *(p1, p2)* defining the edges of the cross-section
-    :vartype facets: list[list[int, int]]
-    :cvar holes: List of points *(x, y)* defining the locations of holes within the cross-section.
-        If there are no holes, provide an empty list [].
-    :vartype holes: list[list[float, float]]
-    :cvar control_points: A list of points *(x, y)* that define different regions of the
-        cross-section. A control point is an arbitrary point within a region enclosed by facets.
-    :vartype control_points: list[list[float, float]]
-    :cvar shift: Vector that shifts the cross-section by *(x, y)*
-    :vartype shift: list[float, float]
-    :cvar perimeter: List of facet indices defining the perimeter of the cross-section
-    :vartype perimeter: list[int]
-    """
+    """Class for defining the geometry of a contiguous section of a single material.
+
+    Provides an interface for the user to specify the geometry defining a section. A method
+    is provided for generating a triangular mesh, transforming the section (e.g. translation,
+    rotation, perimeter offset, mirroring), aligning the geometry to another geometry, and
+    designating stress recovery points.
+
+    :cvar geom: a Polygon object that defines the geometry
+    :vartype geom: shapely.geometry.Polygon
 
-    def __init__(self, control_points, shift):
-        """Inits the Geometry class."""
+    :cvar material: Optional, a Material to associate with this geometry
+    :vartype material: Optional[sectionproperties.pre.Material]
+    """
 
-        self.control_points = control_points
-        self.shift = shift
+    def __init__(
+        self,
+        geom: shapely.geometry.Polygon,
+        material: pre.Material = pre.DEFAULT_MATERIAL,
+    ):
+        """Inits the Geometry class.
+        """
+        if isinstance(geom, MultiPolygon):
+            raise ValueError(f"Use CompoundGeometry(...) for a MultiPolygon object.")
+        if not isinstance(geom, Polygon):
+            raise ValueError(
+                f"Argument is not a valid shapely.geometry.Polygon object: {geom}"
+            )
+        self.geom = geom
+        self.material = material
+        self.control_points = []
+        self.shift = []
         self.points = []
         self.facets = []
         self.holes = []
         self.perimeter = []
+        self._recovery_points = []
+        self.tol = 12 # Represents num of decimal places of precision for point locations
+        self.compile_geometry()
+
+    def _repr_svg_(self):
+        print("sectionproperties.pre.sections.Geometry")
+        print(f"object at: {hex(id(self))}")
+        print(f"Material: {self.material.name}")
+        return self.geom._repr_svg_()
+
+    @staticmethod
+    def from_points(
+        points: List[List[float]],
+        facets: Optional[List[List[int]]] = None,
+        holes: Optional[List[List[float]]] = None,
+        control_points: Optional[List[List[float]]] = None,
```

Comment:
@connorferster, curious if there is ever a use case envisaged where you would want to specify by hand a control_point vs automatically generating it to be within the geometry/section. Potentially seems a little redundant, and one of the benefits to going to shapely was to have them autogenerated vs having to think about specifying them manually?

Reply:
Will create a test for it also.

Reply:
Ah, I see. 

See one of my last replies in #21 as I think there is a way to automatically create the control points within each bit of geometry. In the case of your composite column case, recognising that the concrete part of the section actually contains a hole in reality where the steel is, you can split up/process the section into the constituent bits without any overlaps and then assign the control points automatically. 

If you could get something like that working it almost means the control_point concept is just an internal requirement that is worked out to aid in the meshing process. Then the user doesn't need to worry about it at all (ideal world) .... I'm sure I'm over simplifying things though!!